db/listener_test.cc

```diff
@@ -1087,9 +1086,11 @@ TEST_F(EventListenerTest, OnFileOperationTest) {
   ASSERT_GT(listener->file_closes_.load(), 0);
   ASSERT_GE(listener->file_syncs_.load(), listener->file_syncs_success_.load());
   ASSERT_GT(listener->file_syncs_.load(), 0);
-  ASSERT_GE(listener->file_truncates_.load(),
+  if (true == options.use_direct_io_for_flush_and_compaction){
```

Comment:
Will this ever be true?

Reply:
The default value is `false`, and the case of it being `true` is never tested. Consequently `OnFileTruncateFinish` is not tested. Can we test it?

Reply:
> The default value is `false`, and the case of it being `true` is never tested. Consequently `OnFileTruncateFinish` is not tested. Can we test it?

I'm not sure I can set it to `true` without having segfault in the test in Phabricator. Maybe I'll try testing this function without direct I/O.